priv/signs.json

```diff
@@ -6155,17 +6155,152 @@
     ]
   },
   {
-    "id": "lechmere_green_line",
+    "id": "lechmere_green_line_eastbound",
     "headway_group": "green_trunk",
     "pa_ess_loc": "GLEC",
+    "read_loop_offset": 30,
+    "text_zone": "e",
+    "audio_zones": ["e"],
+    "type": "realtime",
+    "source_config": [
+      [
+        {
+          "stop_id": "70501",
+          "direction_id": 1,
+          "headway_direction_name": "Union Square",
+          "routes": ["Green-E"],
+          "platform": null,
+          "terminal": false,
+          "announce_arriving": true,
+          "announce_boarding": false
+        }
+      ]
+    ]
+  },
+  {
+    "id": "lechmere_green_line_westbound",
+    "headway_group": "green_trunk",
+    "pa_ess_loc": "GLEC",
+    "read_loop_offset": 90,
+    "text_zone": "w",
+    "audio_zones": ["w"],
+    "type": "realtime",
+    "source_config": [
+      [
+        {
+          "stop_id": "70502",
+          "direction_id": 0,
+          "headway_direction_name": "Heath Street",
+          "routes": ["Green-E"],
+          "platform": null,
+          "terminal": false,
+          "announce_arriving": true,
+          "announce_boarding": false
+        }
+      ]
+    ]
+  },
+  {
+    "id": "lechmere_green_line_mezzanine",
+    "headway_group": "green_trunk",
+    "pa_ess_loc": "GLEC",
+    "read_loop_offset": 120,
+    "text_zone": "m",
+    "audio_zones": ["m"],
+    "type": "realtime",
+    "source_config": [
+      [
+        {
+          "stop_id": "70501",
+          "direction_id": 1,
+          "headway_direction_name": "Union Square",
+          "routes": ["Green-E"],
+          "platform": null,
+          "terminal": false,
+          "announce_arriving": true,
+          "announce_boarding": false
+        },
+        {
+          "stop_id": "70502",
+          "direction_id": 0,
+          "headway_direction_name": "Heath Street",
+          "routes": ["Green-E"],
+          "platform": null,
+          "terminal": false,
+          "announce_arriving": false,
+          "announce_boarding": false
+        }
+      ]
+    ]
+  },
+  {
+    "id": "union_sq_east",
```

Comment:
You know, come to think of it, it seems kind of odd to me that we have separate east, west, and mezzanine zones at Union Sq since I'd expect all signs to display the same data: trains towards Heath St. We don't generally show arrival predictions for when trains will get to the terminal.

We've asked Andrew to consolidate zones before (and break apart zones), so that might be worth considering here.

As long as they're separate zones, you might want to do something like: give them all the same `source_config`, and give one of the zones `"audio_zones": ["e", "w", "m"]`, and the others `[]`. That way, it won't be reading the same information all the time, and when it does read it out, it will send the message like `sta=GUNS111000` as a single request, which should make sure they play in sync.

Reply:
As for what that `source_config` should be, that's still an open question. We haven't decided yet on the realtime team exactly what stop IDs we'll use. Static GTFS uses `70503` for arrivals and `70504` for departures, but operationally there are two platforms there and either platform could handle the train pulling in (and then subsequent departure).

So you might want a `source_config` of two entries, `70503` and `70504` both in the `0` direction, and so it would combine the predictions out of either platform.

However, we're also still considering an approach like we do for Alewife, Forest Hills, and Oak Grove, where trains are predicted _to_ `70503` and reverse predictions would be _from_ `70504`, but then when the train is actually _at_ the platform there's be yet two more stop IDs `Union Square-01` and `Union Square-02`. And in that case, you'd want all four stop IDs in your `source_config` list!

I think for now it's good to just go with the first approach (`70503` and `70504` in the `0` direction), and if we go with `Union Square-01` and `Union Square-02`, we'll let you know and you can add them as well.

Reply:
Thank you for the clarification and suggestions! Super helpful especially since I definitely am still getting a handle on how exactly the configs here will play out in real life at the stations.

Re: the audio zones, do you think it might be worth it to have another sign zone have all of the audio zones as well? Just thinking that if only one of the signs has all three audio zones, the audio might play pretty infrequently. Let me know your thoughts there!

Reply:
The read loop triggers the sign to read once every 4 minutes, reset by any "interrupting reads" from things like the sign going to "now arriving" or "stopped n stops away". That's the value used by all signs throughout the system. If the signs at Union Square are all configured to show the same information (which I'm guessing they should, since they [all look close to each other](https://upload.wikimedia.org/wikipedia/commons/thumb/a/a4/Union_Square_station_construction_%283%29%2C_December_2021.JPG/1200px-Union_Square_station_construction_%283%29%2C_December_2021.JPG) and the trains only go one way), then conceptually they're just a single zone and should read at once / 4 minutes.

Adding audio zones to another sign will either have them start out on a sort of heartbeat cadence, depending on the `read_loop_offset`, as they'd each read every 4 minutes staggered by their offset. But then an "interrupting read" would happen and they'd all be on the same every-4-minute cycle anyway.

Reply:
Makes sense!

Re: The source config, we'd still want the `text_zone` and `id` to be different from each other, since that's what allows us to specify which signs we're talking to right? At least until we decide if we want to ask Andrew to combine the zones into one and regardless of the fact that they're all showing the same information for stop_id 70503 and 70504

In other words, for now their source configs should be identical except we'd still want separate `id`s (such as `union_sq_west`) and their respective `text_zone` values?

Reply:
Yes, sorry by `source_config` I meant specifically just the data in the `"source_config"` key. I think we call the overall thing the "sign config" maybe? I admit it's a bit confusing. I think the signs should have identical `"source_config"` lists, but yes, the `"id"`, `"text_zone"`, and `"audio_zones"` would be different.

Reply:
Going back to the topic of the announcements at the new Union Sq station. I know that at other terminals, we keep the signs in headway mode although I'm unsure if that's only where there is a train yard present. Would that be the case at Union Sq, or would we want to announce departures or something since there's no train yard (I think) at Union Sq and I'm guessing that in that case, the trains just turn around and go the other way?

Reply:
We've currently set expectations as the signs there only being in headway mode. There is a yard at Union Square station (though inbound and not *past* the station). So there's a potential for trains to arrive at Union Sq, offload passengers, and then start heading back empty and then pull into the yard.

Furthermore, we don't really know operationally what will be going on there. After a train arrives and offloads passengers, it will probably wait some amount of time, but we don't know if that will be a consistent amount of time or very accurately predictable. And sadly we don't have "train sheets" like we do for the heavy rail where the dispatchers tee up the departures and send us messages "delay the trip departure by 3 minutes" or "planning to leave one minute early". There is the Glides program here in CTD that's working on, among other things, just that, though! So hopefully someday we'll have good departure predictions.

Reply:
I see, so sounds like keeping the zones as configured to not provide arrival/departure messages is correct given the current expectations and state of the system 😄